### DISCUSSION TITLE
Use SkPico as synth

### DISCUSSION THREAD

**Discussion by pedja77** [frntc/SIDKick-pico#14] [Q&A] [answered]
Hi,
I really like this project. I wonder is it possible to use another pi pico for midi input and maybe some other controls and connect SKpico and play it like midi controlled synth?

**Comment by frntc** [comment]
You may have seen the big brother of SIDKick pico: the Teensy-based SID-replacement. It provides a MIDI interface, and I was considering building a MIDI interface with a Pico -- then there would be software on the C64 providing a SID MIDI synth.

**Reply by pedja77** [reply]
Thank you for answer. I'm aware of picos big brother. I'm planing on building a rig for live performance and sid sounds will be quite welcome. But bringing full c64's along with other equipment will be quite a problem logistics wise.
In theory, right me if i'm wrong, one can interface one pico (midi in) to another {with SikKick firmware) directly, without voltage level converters used in SidKick pico? If not directly, then trough shift registers (like someone did with arduino and real sid). All that is needed are pinout mapping that pico - sid sidkick uses and a way to configure firmware manually. Maybe also some changes in firmware.
That is my idea in a nutshell.

**Reply by frntc** [reply, marked as answer]
You could interface two Picos, but you would probably want to do quite a few chances (e.g. the firmware relies on the C64 clock signal which you would not have). I think the easiest solution would be to take the SKpico firmware, remove the bus-interface part and replace that particular part with a MIDI routine (which also does the mapping from MIDI commands to SID register writes).

**Reply by pedja77** [reply]
Thank you for time to answer.
